ksqldb-engine/src/main/java/io/confluent/ksql/engine/QueryCleanupService.java

```diff
@@ -111,35 +114,55 @@ public String getAppId() {
       return appId;
     }
 
+    public String getQueryId() {
+      return queryId;
+    }
+
     @Override
     public void run() {
-      try {
-        final Path pathName = Paths.get(stateDir + "/" + appId);
-        final File directory = new File(String.valueOf(pathName.normalize()));
-        if (directory.exists()) {
-          FileUtils.deleteDirectory(directory);
-          LOG.warn("Deleted local state store for non-existing query {}. "
-                  + "This is not expected and was likely due to a "
-                  + "race condition when the query was dropped before.",
-              appId);
+      if (queryId.equals("")) {
```

Comment:
nit: can you define a constant for this instead of hard-coded `""` around the codebase so that it's clear what the meaning of this is? Otherwise I think it might be a bit confusing for someone who isn't familiar with the shared runtimes/NamedTopology stuff since it's not obvious why the `queryId` would ever be empty, eg
```
public static final String DEDICATED_RUNTIME_QUERY = "";
```

Reply:
I can use optional, empty for dedicated runtimes

Reply:
Whatever you prefer, as long as it's clear enough to someone who might read this code in the future without detailed knowledge of the shared runtime architecture